executor.go

```diff
@@ -161,6 +175,12 @@ func (e *ExecutorBuilder) NewCmdExecutor(ctx context.Context) cmd.Executor {
 			return
 		}
 
+		shutdownFn, err := kotel.Register(ctx, logger, cfg)
```

Comment:
the otel registration should be done at the `MetricsAndTracesRegister` collaborator. The implementation of the `MetricsAndTraces` could contain an exported property for the CloseFunc.